Sources/MongoSwift/ReadWriteConcern.swift

```diff
@@ -114,3 +113,166 @@ extension ReadConcern: Equatable {
         return lhs.level == rhs.level
     }
 }
+
+/// A class to represent a MongoDB write concern.
+public class WriteConcern: Codable {
+
+    /// A pointer to a mongoc_write_concern_t
+    internal var _writeConcern: OpaquePointer?
+
+    /// An option to request acknowledgement that the write operation has propagated to specified mongod instances.
+    public enum W: Codable, Equatable {
+        /// Specifies the number of nodes that should acknowledge the write. MUST be greater than or equal to 0.
+        case number(Int32)
+        /// Indicates a tag for nodes that should acknowledge the write. 
+        case tag(String)
+        /// Specifies that a majoirty of nodes should acknowledge the write.
+        case majority
+
+        public init(from decoder: Decoder) throws {
+            let container = try decoder.singleValueContainer()
+            if let wTag = try? container.decode(String.self) {
+                self = .tag(wTag)
+            } else {
+                let wNumber = try container.decode(Int32.self)
+                if wNumber == MONGOC_WRITE_CONCERN_W_MAJORITY {
```

Comment:
Is it worth handling other [numeric constants](https://github.com/mongodb/mongo-c-driver/blob/1.10.2/src/libmongoc/src/mongoc/mongoc-write-concern.h#L32) here?